src/sidebar/store/modules/filters.js

```diff
@@ -0,0 +1,231 @@
+import { createSelector } from 'reselect';
+
+import { actionTypes } from '../util';
+
+/**
```

Comment:
These Focus-related typedefs mostly already existed before (in `selection`), with a couple of changes:

* `Focus` used to be `FocusState`
* `FocusState` is new